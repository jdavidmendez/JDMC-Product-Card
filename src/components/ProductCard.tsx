import styles from "./../styles/styles.module.css";
import { createContext } from "react";
import {
  ProductCardProps,
  ProductContextProps,
} from "../interfaces/interfaces";
import { useProduct } from "../hooks/useProduct";
import React from "react";

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export const ProductCard = ({
  children,
  product,
  className,
  style,
  onChange,
  value,
  initialValues,
}: ProductCardProps) => {
  const { counter, increaseBy, maxCount, isMaxReached, reset } = useProduct({ onChange, product, value, initialValues });
  return (
    <Provider
      value={{
        counter,
        increaseBy,
        maxCount,
        product,
      }}
    >
      <div className={`${styles.productCard} ${className}`} style={style}>
        {children({
          count: counter,
          isMaxReached,
          maxCount: initialValues?.maxCount,
          product,
          increaseBy,
          reset
        })}
      </div>
    </Provider>
  );
};
