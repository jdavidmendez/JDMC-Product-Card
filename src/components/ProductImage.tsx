import { useContext } from 'react';
import { ProductContext } from './ProductCard';
import noImage from './../assets/no-image.jpg';
import styles from './../styles/styles.module.css';
import React from "react";

export interface ProductImageProps {
    img?: string;
    className?: string;
    style?: React.CSSProperties;
}
export const ProductImage = ({ img, className,style }: ProductImageProps) => {
    const { product } = useContext(ProductContext);
    let imgToshow: string;
    imgToshow = img ? img : product.img ? product.img : noImage;
    return (
        <img className={`${styles.productImg} ${className}`} src={imgToshow} alt="Coffe mug"  style={style}/>
    )
}