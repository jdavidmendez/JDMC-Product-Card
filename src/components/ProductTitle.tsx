import { useContext } from "react";
import { ProductContext } from "./ProductCard";
import styles from "./../styles/styles.module.css";
import React from "react";

export interface productTitleProps {
  title?: string;
  className?: string;
  style?: React.CSSProperties;
}

export const ProductTitle = ({
  title,
  className,
  style,
}: productTitleProps) => {
  const { product } = useContext(ProductContext);
  return (
    <span className={`${styles.productDescription} ${className}`} style={style}>
      {title ? title : product.title}
    </span>
  );
};
  