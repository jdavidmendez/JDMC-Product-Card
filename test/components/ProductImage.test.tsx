import renderer from 'react-test-renderer';
import { ProductImage } from '../../src/components/ProductImage';
import React from 'react';
import ProductCard from '../../src/components';
import { product2 } from '../data/products';

describe('pruebas en ProductImage', () => {
    test('debe mostrarse correctamente el componente', () => {
        const wrapper = renderer.create(
            <ProductImage img='https://gatito.com/image.jpg' />
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    })

    test('debe mostrar el componente con el nombre del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
})