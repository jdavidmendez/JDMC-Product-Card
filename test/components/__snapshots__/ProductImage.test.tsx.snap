// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en ProductImage debe mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffe mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`pruebas en ProductImage debe mostrarse correctamente el componente 1`] = `
<img
  alt="Coffe mug"
  className="productImg undefined"
  src="https://gatito.com/image.jpg"
/>
`;
